core/src/main/scala/kafka/zk/ZkMigrationClient.scala

```diff
@@ -211,19 +230,21 @@ class ZkMigrationClient(zkClient: KafkaZkClient) extends MigrationClient with Lo
     }
   }
 
-  override def readAllMetadata(batchConsumer: Consumer[util.List[ApiMessageAndVersion]],
-                               brokerIdConsumer: Consumer[Integer]): Unit = {
+  override def readAllMetadata(
+    batchConsumer: Consumer[util.List[ApiMessageAndVersion]],
+    brokerIdConsumer: Consumer[Integer]
+  ): Unit = wrapZkException {
```

Comment:
I think it would be better to have the `wrapZkException` blocks inside `migrateTopics`, `migrateBrokerConfigs`, `migrateClientQuotas`, `migrateProducerId`. Then we can have a simple mental rule like: if you do something with zkClient, you need to wrap it. And not have to analyze control flow.